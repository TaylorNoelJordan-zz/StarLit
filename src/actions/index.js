export const setUser = (user) => ({
    type: 'SET_USER',
    user
});

export const clearUser = () => ({
    type: 'CLEAR_USER'
});

export const hasErrored = errorMsg => ({
    type: 'HAS_ERRORED',
    errorMsg
});

export const setVerdict = (status) =>({
    type: 'SET_VERDICT',
    status
})