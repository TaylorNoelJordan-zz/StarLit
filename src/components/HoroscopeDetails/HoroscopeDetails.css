.horoscope-details-display {
    align-items: center;
    display: flex;
    flex-direction: column;
    font-size: 1.3em;
    margin: 20px;
    margin-top: 0px;
}

.horoscope-details {
    color: #9cd8cd;
    margin: 20px;
    padding: 15px;
    width: 600px;
}