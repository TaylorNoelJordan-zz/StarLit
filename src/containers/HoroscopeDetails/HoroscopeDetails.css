@import url('https://fonts.googleapis.com/css?family=Darker+Grotesque|Lobster+Two|Poiret+One&display=swap');


.horoscope-details-display {
    align-items: center;
    display: flex;
    flex-direction: column;
    font-family: 'Poiret One', cursive;
    font-size: 1.3em;
    margin-top: 0px;
}

.horoscope-details {
    color: #9cd8cd;
    /* margin: 20px; */
    /* padding: 15px; */
    width: 600px;
}

.results-heading {
    margin-top: 20px;
}

.start-over-btn {
    border-radius: 15px;
    box-shadow: 0px 0px 8px #fff;
    font-size: 18px;
    font-weight: bold;
    height: 50px;
    margin: 15px;
    opacity: 0.7;
    width: 130px;
  }
  
  .start-over-btn:hover {
    box-shadow: 0px 0px 15px #fff;
    color: #EB4D8D;
  }

  .yasss-btn {
    color: white;
    font-weight: bold;
    margin: 5px;
  }

  .verdict-planet {
    height: 45px;
    margin: 5px;
  }