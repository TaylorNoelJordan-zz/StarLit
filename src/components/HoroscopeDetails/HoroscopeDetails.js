import React from 'react';
import AstroSign from '../AstroSign/AstroSign';
import { connect } from 'react-redux';
import { loadComplete } from '../../actions'
import './HoroscopeDetails.css';

const HoroscopeDetails = (props) => {
    const { sign, name, dateRange, message, isLoading } = props;
    const loadingGif = 'https://media2.giphy.com/media/xTkcEQACH24SMPxIQg/giphy.gif';
    const loading = (
      <section className="loading-gif-container">
        <img src={loadingGif} alt="loading-gif" className="loading-gif" />
      </section>
    );
    return (
        <section className='horoscope-details-display'>
        {/* {isLoading && loading} */}
            <AstroSign 
                sign={sign}
                dateRange={dateRange}
            />
            <article className='horoscope-details'>
            <h2>A-yo {name},</h2>
                <p>{message}</p>
            </article>
        </section>
    )
}

const mapStateToProps = state => ({
    sign: state.user.sign,
    name: state.user.name,
    dateRange: state.user.date_range,
    message: state.user.description,
    isLoading: state.isLoading
});

const mapDispatchToProps = dispatch => ({
  endLoading: () => dispatch(loadComplete())
})

export default connect(mapStateToProps, mapDispatchToProps)(HoroscopeDetails);