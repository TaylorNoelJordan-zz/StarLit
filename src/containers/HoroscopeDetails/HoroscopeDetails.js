import React from 'react';
import AstroSign from '../AstroSign/AstroSign';
import { Link, Redirect } from 'react-router-dom';
import { clearUser } from '../../actions';
import { connect } from 'react-redux';
import FadeIn from 'react-fade-in';
import PropTypes from 'prop-types';
import happyPlanet from '../../images/033-planet-7.svg';
import sadPlanet from '../../images/028-planet-4.svg';
import './HoroscopeDetails.css';

export const HoroscopeDetails = (props) => {
    const yesMessage = (
            <>
                <img src={happyPlanet} alt='smiling purple planet' className='verdict-planet'/>
                <p>The stars have aligned and the heavens have parted! The cosmos give permission for an evening of boozy indulgence, 
                    with the condition that the vibes stay as high as possible. Stay safe and have fun! 
                    Need some advice on what to wear and who to link up with?</p>
                    <Link to='/details' className='yasss-btn'>YASSSS</Link>
            </>
    )

    const noMessage = (
        <>
            <img src={sadPlanet} alt='sleeping purple planet' className='verdict-planet'/>
            <p>Outlook not so good. Looks like the planetary alignment is calling for a little more detox, a little less retox. 
                Take tonight as an opportunity to tune inward, face that scary stuff youâ€™ve been putting off, do a face mask and 
                get some deep sleep. Enjoy a night in and check back tomorrow.</p>
        </>
    )

    const { name, message, verdict, clearUser } = props;
    return (
        <FadeIn>
            <section className='horoscope-details-display'>
                <AstroSign />
                <article className='horoscope-details'>
                <h2 className='results-heading'>A-yo {name},</h2>
                    <p>{message}</p>
                <h2 className='results-heading'>The Verdict:</h2>
                    {verdict ==='positive' ? yesMessage : noMessage}
                </article>
                {!name && <Redirect to='/'/>}
                <Link to='/'>
                    <button 
                        className='start-over-btn'
                        onClick={clearUser}>
                            Start Over
                    </button>
                </Link>
            </section>
        </FadeIn>
    )
}

export const mapStateToProps = state => ({
    user: state.user,
    name: state.user.name,
    message: state.user.description,
    verdict: state.verdict
});

export const mapDispatchToProps = dispatch => ({
    clearUser: () => dispatch(clearUser())
})


HoroscopeDetails.propTypes = {
    user: PropTypes.object,
    name: PropTypes.string,
    message: PropTypes.string,
    verdict: PropTypes.string
}



export default connect(mapStateToProps, mapDispatchToProps)(HoroscopeDetails);