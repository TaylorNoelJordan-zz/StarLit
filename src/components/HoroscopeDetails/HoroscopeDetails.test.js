import React from 'react';
import { HoroscopeDetails, mapStateToProps, mapDispatchToProps }from './HoroscopeDetails'
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import { clearUser } from '../../actions';

describe('HoroscopeDetails', () => {
    let wrapper;
    let mockState;

    beforeEach(() => {
        mockState = {
            user: {
                name: 'Eleven',
                sign: 'Scorpio',
                description: 'The demagorgen didn\'t beat you, and neither shall the mind flayer. Keep a clear head tonight... You may be responsible for saving all your friends.',
                color: 'gray',
                mood: 'on',
                date_range: ''
            },
            verdict: ''
        }
        const { user, name, sign, verdict } = mockState
        wrapper = shallow(<HoroscopeDetails props={mockState}/>);
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should route to more details with affirmative results', () => {
        expect(wrapper.find(Link).props().to).toBe('/details');
    });

    describe('mapStateToProps', () => {
        it('should return corresponding portions of state', () => {
            const mockState = {
                user: {
                    name: 'Eleven',
                    sign: 'Scorpio',
                    description: 'The demagorgen didn\'t beat you, and neither shall the mind flayer. Keep a clear head tonight... You may be responsible for saving all your friends.',
                    color: 'gray',
                    mood: 'on',
                    date_range: ''
                },
                verdict: ''
            }

            const expected = {
                    name: mockState.user.name,
                    sign: mockState.user.sign,
                    message: mockState.user.description,
                    dateRange: mockState.user.date_range,
                    verdict: ''
            }

            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected)
        });
    });

    describe('mapDispatchToProps', () => {
        let mockDispatch;

        beforeEach(() => {
            mockDispatch = jest.fn();
        });

        it('should call dispatch with clearUser', () => {
            const actionToDispatch = clearUser();

            const mappedProps = mapDispatchToProps(mockDispatch)
            mappedProps.clearUser()

            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
        })
    })
})