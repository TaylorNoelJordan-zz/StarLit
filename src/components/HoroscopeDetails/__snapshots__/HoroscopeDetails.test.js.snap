// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoroscopeDetails should match the snapshot 1`] = `ShallowWrapper {}`;
